### ISSUE TITLE
Backbone-relational + Browserify

### ISSUE THREAD

**Issue by davekiss** [PaulUithol/Backbone-relational#528]
Has anyone been able to successfully get Backbone-relational working with Browserify? I'm at 3 days of hacking with no luck. Been getting all kinds of errors, but these are the most common (they are untrue):

`Relation=s: relatedModel does not inherit from Backbone.RelationalModel (s).`
`collectionType must inherit from Backbone.Collection`

I've already seen `addModelScope( models );` but it doesn't seem to solve the issues. Would I need to use that in every module I create? I was trying something like this:

```
var models      = window.Namespace.models      = window.Namespace.models      || {};
var collections = window.Namespace.collections = window.Namespace.collections || {};
Backbone.Relational.store.addModelScope( models );

models.Course = require('../models/course_model');
var course = new models.Course( Namespace.courses[0] );
```

I'm guessing this is some kind of scoping issue but I am at a loss as to how to get this set up properly in this type of environment. If you've done it successfully, can you post your Browserify recipe or strategy?

**Comment by PaulUithol** [comment]
Quite sure other people have had better luck with this; see #502 for example. Don't use it myself, so unfortunately can't help out right away.

**Comment by Narsul** [comment]
Hey. Well I have a file ```app_loader.js``` which is my entry point to entire app.
In it I'm doing all the hacks to support some nasty plugins etc.

Try to browserify this:
```
var $ = require('jquery');                                  // jQuery
var Backbone = require('backbone');                         // Backbone
Backbone.$ = $;                                             // Setting jQuery as Backbone's DOM library
require('backbone-relational');                             // Backbone relational plugin

var models = {};
Backbone.Relational.store.addModelScope(models);

models.Person = Backbone.RelationalModel.extend({
    relations: [{
        type: 'HasOne',
        key: 'address',
        relatedModel: 'Address',
        keySource: 'addressId'
    }]
});
models.Address = Backbone.RelationalModel.extend({});

var john = models.Person.findOrCreate({id: 1, firstName: 'John', lastName: 'Smith', addressId: 2});
var home = models.Address.findOrCreate({id: 2, name: 'home', street: '...'});

console.log(john.toJSON());
```

**Comment by davekiss** [comment]
Where I ran in to trouble was when I was including the models in their own separate files and using module.exports to make them available in a components file. Still wasn't able to get it to work that way.

**Comment by quannt** [comment]
Any update on this? Did you manage to fix this in the end? @davekiss

**Comment by bpatram** [comment]
A resolution to this annoyance should be fixed with this PR: #559
